save-common/src/commonMain/kotlin/org/cqfn/save/core/config/TestConfig.kt

```diff
@@ -71,61 +73,49 @@ data class TestConfig(
     fun getAllTestConfigs(): List<TestConfig> {
         return listOf(this) + this.childConfigs.flatMap { it.getAllTestConfigs() }
     }
-    
+
     /**
-     * Merge parent configurations with current and prolong it for all child configs
+     * filtering out general configs
      */
-    fun merge() {
-        logDebug("Start merging configs for ${this.location}")
-        val parentConfigs = parentConfigs(withSelf = true).toList().asReversed()
-        mergeConfigList(parentConfigs)
-        mergeChildConfigs()
-    }
+    fun pluginConfigsWithoutGeneralConfig() = pluginConfigs.filterNot { it is GeneralConfig }
 
-    // Merge list of configs pairwise
-    private fun mergeConfigList(configList: List<TestConfig>) {
-        if (configList.size == 1) {
-            return
-        }
-        val pairs = configList.zipWithNext()
+    /**
+     * Merge parent list of plugins with the current list
+     *
+     * @return merged test config
+     */
+    fun mergeConfigWithParents(): TestConfig {
+        logDebug("Start merging configs for ${this.location}")
 
-        pairs.forEach { (parent, child) ->
-            child.mergeChildConfigWithParent(parent)
+        this.parentConfigs().toList().forEach { parentConfig ->
```

Comment:
Is it the correct logic? We should start merging from the root config.
Also, if we are starting from the middle, then we should traverse parents once, and then go down from the provided node

Reply:
discussed